lib/jekyll-admin/server.rb

```diff
@@ -75,7 +75,11 @@ def write_file(path, content)
       File.open(path, "wb") do |file|
         file.write(content)
       end
-      site.process
+      if ENV["RACK_ENV"]
+        site.process
+      else
+        site.read
+      end
```

Comment:
What if, instead, we disable watch and manually call `site.process` (which should then be blocking). As of right now, each write will require two reads (and I believe now, two full processes).